frontend/src/app/core/setup/globals/global-listeners/hover-card-trigger.service.ts

```diff
@@ -49,31 +60,48 @@ export class HoverCardTriggerService {
       e.stopPropagation();
       // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
       const el = e.target as HTMLElement;
-      if (el) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        const turboFrameUrl = el.getAttribute('data-hover-card-url');
-
-        if (!turboFrameUrl) {
-          return;
-        }
-
-        this.opModalService.show(
-          HoverCardComponent,
-          this.injector,
-          { turboFrameSrc: turboFrameUrl, event: e },
-          true,
-        ).subscribe((previewModal) => {
-          this.modalElement = previewModal.elementRef.nativeElement as HTMLElement;
-          void previewModal.reposition(this.modalElement, el);
-        });
+      if (!el) { return; }
+
+      if (this.previousTarget && this.previousTarget === el) {
+        // Re-entering the trigger counts as hovering over the card:
+        this.mouseInModal = true;
+        // But we will not re-render the same card, abort here
+        return;
       }
+
+      // Hovering over a new target. Close the old one (if any).
+      this.close(true);
+      this.previousTarget = el;
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const turboFrameUrl = el.getAttribute('data-hover-card-url');
+      if (!turboFrameUrl) { return; }
+
+      // When set in an angular component, the url attribute might be wrapped in additional quotes. Strip them.
+      const cleanedTurboFrameUrl = turboFrameUrl.replace(/^"(.*)"$/, '$1');
```

Comment:
The reason this happens is because you use the inputs that are used for components in the AvatarHelper, which is not too useful for services.

These often get read in this helper function:
https://github.com/opf/openproject/blob/c757e6f60e68c995f6c17db52fa379fb5b62aeb7/frontend/src/app/shared/components/dataset-inputs.ts#L80

and in the end, it's JSON. Of course replacing it might be fine, but `JSON.parse` would also work. What I don't understand: You get the `hoverCardUrl` in the principal component, so it's parsed correctly there. Shouldn't this be used to render the avatar? In case of work packages, it seems a plain data-attribute is already used.

Are these lines still necessary?

Reply:
Yes, we actually don't need these lines anymore. This is a left over from a time earlier during development when I applied the trigger to the principal-tag directly. This was done to support having a popover on both the avatar and the name (and possibly the gap in-between them). Back then, these data-attributes were read from the principal angular tag by the trigger service - breaking it.

Since the trigger is now attached to the avatar image tag, encountering a broken data-url is no longer an issue and this code can be removed 🧹